### ISSUE TITLE
[Feature Request]: Use SwayOSD for volume and brightness

### ISSUE THREAD

**Issue by TurtleMeds** [prasanthrangan/hyprdots#1738]
### Description

Use the `swayosd-git` package from the aur to make changing volume and brightness much more intuitive and feel less hacky.

### Use Case

It would look much better and more polished.

### Alternatives

it is also possible to make your own with something like eww or ags.

### Additional Information

![image](https://github.com/user-attachments/assets/32e6fc87-a49b-4634-bb51-f3686c1467b6)
![image](https://github.com/user-attachments/assets/8fdb44b4-182a-4dcf-b994-3eae8650c816)

**Comment by kRHYME7** [comment]
Unfortunately, compiling and installing swayOSD on low-end devices takes a significant amount of time. And I'm afraid would cause issues. We can add this in the optional package though

**Comment by rubiin** [comment]
That would be great. I personally like the design compared to what we have now.

**Comment by kRHYME7** [comment]
I converted the binds BTW while testing
```
# Volume controls
#bind=,XF86AudioRaiseVolume,exec,swayosd-client --output-volume raise
#bind=,XF86AudioLowerVolume,exec,swayosd-client --output-volume lower
#bind=,XF86AudioMute,exec,swayosd-client --output-volume mute-toggle

# Source volume toggle mute
#bind=,XF86AudioMicMute,exec,swayosd-client --input-volume mute-toggle

# Custom volume values
#bind=,XF86AudioRaiseVolume,exec,swayosd-client --output-volume 15
#bind=,XF86AudioLowerVolume,exec,swayosd-client --output-volume -15

# Max volume
#bind=,XF86AudioRaiseVolume,exec,swayosd-client --output-volume raise --max-volume 120
#bind=,XF86AudioLowerVolume,exec,swayosd-client --output-volume lower --max-volume 120

# Custom volume values with device
#bind=,XF86AudioRaiseVolume,exec,swayosd-client --output-volume +10
#bind=,XF86AudioLowerVolume,exec,swayosd-client --output-volume -10

# Caps Lock
#bindr=,Lock,exec,swayosd-client --caps-lock

# Brightness controls
#bind=,XF86MonBrightnessUp,exec,swayosd-client --brightness raise
#bind=,XF86MonBrightnessDown,exec,swayosd-client --brightness lower

# Custom brightness values
#bind=,XF86MonBrightnessUp,exec,swayosd-client --brightness +10
#bind=,XF86MonBrightnessDown,exec,swayosd-client --brightness -10
```

**Comment by rubiin** [comment]
I did too. Few buttons like mute and volumes just to see the UI

**Comment by rubiin** [comment]
I will try to swap the hyprkeybinds with this one this weekend . Will update by then

**Comment by rubiin** [comment]
![image](https://github.com/user-attachments/assets/7cf9c63c-9376-4b94-8518-ca7eb7d10437)

@kRHYME7

**Comment by TurtleMeds** [comment]
> Did you happen to make caps_lock work? 

Works out of the box for me

**Comment by rubiin** [comment]
same

**Comment by kRHYME7** [comment]
Does their repo have the bin files? I guess we ask for feat req there and someone can make an aur pkg.

**Comment by kRHYME7** [comment]
What's the reason for not shipping bin? Should it be compiled specifically for the system?

**Comment by rubiin** [comment]
yeah. They dont have done any release as far I can see. You build with cargo lol

**Comment by kRHYME7** [comment]
Maybe we should try to request then. Maybe they can provide any reasons why they're not shipping a bin.

I'm using chaotic aur, still no one packages it to bin.

**Comment by rubiin** [comment]
https://github.com/chaotic-aur/packages/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen lets create one here for swayosd-git.
We will need users upvoting for quick i guess

**Comment by rubiin** [comment]
@kRHYME7 @TurtleMeds created a request here, be sure to comment there so the devs can release a bin sooner
https://github.com/chaotic-aur/packages/issues/3303

Then we can get going with this feature

**Comment by Technetium1** [comment]
The package has been added. For as long as we provide it build logs will be available at https://builds.garudalinux.org/repos/chaotic-aur/logs/swayosd-git.log
Eventually after transition to v4, logs will be available here https://aur.chaotic.cx/status
